frontend/src/api/k8s/templates.ts

```diff
@@ -39,18 +46,43 @@ export const assembleServingRuntimeTemplate = (
   };
 };
 
-export const listTemplates = async (
-  namespace?: string,
-  labelSelector?: string,
-): Promise<TemplateKind[]> => {
-  const queryOptions = {
-    ...(namespace && { ns: namespace }),
-    ...(labelSelector && { queryParams: { labelSelector } }),
-  };
-  return k8sListResource<TemplateKind>({
-    model: TemplateModel,
-    queryOptions,
-  }).then((listResource) => listResource.items);
+export const useTemplates = (namespace?: string): CustomWatchK8sResult<TemplateKind[]> => {
+  const modelServingEnabled = useModelServingEnabled();
+  const customServingRuntimesEnabled = useCustomServingRuntimesEnabled();
+
+  const initResource: WatchK8sResource | null =
+    namespace && modelServingEnabled
+      ? {
+          isList: true,
+          groupVersionKind: groupVersionKind(TemplateModel),
+          namespace,
+        }
+      : null;
+
+  const [templatesData, loaded, error] = useCustomK8sWatchResource<TemplateKind[]>(
+    initResource,
+    TemplateModel,
+  );
+
+  const templates = React.useMemo(
+    () =>
+      customServingRuntimesEnabled
+        ? templatesData
+        : templatesData.filter(
```

Comment:
Isn't it possible that `templatesData` is `undefined` here if `initResource == null` ?

Reply:
yes. I will handle that case.